corehq/ex-submodules/casexml/apps/case/tests/test_util.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+
+from django.test import SimpleTestCase
+
+from iso8601 import iso8601
+
+from casexml.apps.case.exceptions import PhoneDateValueError
+from casexml.apps.case.util import validate_phone_datetime
+
+
+class ValidatePhoneDatetimeTests(SimpleTestCase):
+
+    def test_datetime_string(self):
+        datetime_string = '2019-08-27T17:50:00.000'
+        result = validate_phone_datetime(datetime_string)
+
+        datetime_ = iso8601.parse_date(datetime_string)
+        assert datetime_ == datetime.datetime(
```

Comment:
This is confusing - why do you need to assert that this is true? Also you could just use the manually created date in the `assertEqual` below if you don't trust the value of `datetime_`

Reply:
My thought here was about using "tests as documentation", to show the reader what is happening. `validate_phone_datetime()` uses `iso8601.parse_date()`. So this `assert` was supposed to be like "Look, this is what we get." ...

But I agree, especially re-reading this, this is confusing. There is no explanation for what that's doing there, and why, and why I used `assert` here and `self.assertEqual()` on the next line. I'll drop this.